examples/CMakeLists.txt

```diff
@@ -1,15 +1,18 @@
 add_executable(main main.cpp)
-
 target_link_libraries(main
     PRIVATE
         everest::log
         Boost::program_options
 )
-
-# needs c++ 14
 target_compile_features(main PRIVATE cxx_std_14)
 
-install(TARGETS main
+add_executable(test_trace test_trace.cpp)
+target_link_libraries(test_trace PRIVATE
+    everest::log
+)
+target_compile_features(test_trace PRIVATE cxx_std_14)
+
+install(TARGETS main test_trace
```

Comment:
I think whilst we are changing things here anyway we should put this behind a CMake flag so the example binaries are not built by default.
Something along the lines of 
```cmake
option(BUILD_EXAMPLES "Build liblog example binaries." OFF)
if(BUILD_EXAMPLES)
    message("Building liblog example binaries.")
    add_subdirectory(src)
else()
    message("Not building libocpp example binaries.")
endif()
```